src/swap/handle_check_address.c

```diff
@@ -146,7 +148,7 @@ static bool swap_check_address_consistency(const check_address_parameters_t *con
         goto out;
     }
 
-    len = strlen(params->address_to_check);
+    len = strnlen(params->address_to_check, ADDRESS_BASE64_LENGTH);
```

Comment:
`strnlen` will return `ADDRESS_BASE64_LENGTH` if the string is longer than ADDRESS_BASE64_LENGTH with will be interpreted as a success:
```suggestion
    len = strnlen(params->address_to_check, ADDRESS_BASE64_LENGTH + 1);
```
If we get `len == ADDRESS_BASE64_LENGTH + 1`, it means that `params->address_to_check[ADDRESS_BASE64_LENGTH] != '\0'`. This way the failure is correctly handled and we do not read outside the bounds (supposing that the buffer size is at least ADDRESS_BASE64_LENGTH + 1).

Reply:
yes got it, thanks for your review